src/Microsoft.DotNet.ExceptionScanning/ExceptionScanner.cs

```diff
@@ -62,8 +62,19 @@ private static ExceptionInfo ScanPlatformNotSupported(IMethodDefinition method,
         {
             const int maxNestingLevel = 3;
 
-            if (method is Dummy || method.IsAbstract)
+            if (method is Dummy
+                || method.IsAbstract 
+                || method.Attributes.Any(attrb => attrb.FullName().Equals("System.Runtime.CompilerServices.IntrinsicAttribute")))
+            {
+                return ExceptionInfo.DoesNotThrow;
+            }
+
+            // Not all intrisincs are marked with attributes
+            var docId = method.DocId();
+            if (docId.StartsWith("M:System.Runtime.Intrinsics") || docId.StartsWith("M:System.ByReference"))
```

Comment:
I considered adding a command-line option to pass a regex but for now I am postponing that for the moment (I want to see more cases before making the choice).